lib/src/core.d.ts

```diff
@@ -0,0 +1,128 @@
+export declare const GetMonitorPhysicalHeight: (monitor: number) => number;
+export declare const GetMonitorName: (monitor: number) => string;
+export declare const GetClipboardText: () => string;
+export declare const SetClipboardText: (text: string) => void;
+export declare const ShowCursor: () => void;
+export declare const HideCursor: () => void;
+export declare const IsCursorHidden: () => boolean;
+export declare const EnableCursor: () => void;
+export declare const DisableCursor: () => void;
+export declare const ClearBackground: (color: Color) => void;
+export declare const BeginDrawing: () => void;
+export declare const EndDrawing: () => void;
+export declare const BeginMode2D: (camera: Camera2D) => void;
+export declare const EndMode2D: () => void;
+export declare const BeginMode3D: (camera: Camera3D) => void;
+export declare const EndMode3D: () => void;
+export declare const BeginTextureMode: (target: RenderTexture2D) => void;
+export declare const EndTextureMode: () => void;
+export declare const GetMouseRay: (mousePosition: Vector2, camera: Camera) => Ray;
+export declare const GetWorldToScreen: (position: Vector3, camera: Camera) => Vector2;
+export declare const GetCameraMatrix: (camera: Camera) => Matrix;
+export declare const SetTargetFPS: (fps: number) => void;
+export declare const GetFPS: () => number;
+export declare const GetFrameTime: () => number;
+export declare const GetTime: () => number;
+export declare const ColorToInt: (color: Color) => number;
+export declare const ColorNormalize: (color: Color) => Vector4;
+export declare const ColorToHSV: (color: Color) => Vector3;
+export declare const ColorFromHSV: (hsv: Vector3) => Color;
+export declare const GetColor: (hexValue: number) => Color;
+export declare const Fade: (color: Color, alpha: number) => Color;
+export declare const SetConfigFlags: (flags: number) => void;
+export declare const SetTraceLogLevel: (logType: number) => void;
+export declare const SetTraceLogExit: (logType: number) => void;
+export declare const SetTraceLogCallback: (callback: TraceLogCallback) => void;
+export declare const TraceLog: (logType: number, text: string, ...args: any) => void;
+export declare const TakeScreenshot: (fileName: string) => void;
+export declare const GetRandomValue: (min: number, max: number) => number;
+export declare const FileExists: (fileName: string) => boolean;
+export declare const IsFileExtension: (fileName: string, ext: string) => boolean;
+export declare const GetExtension: (fileName: string) => string;
+export declare const GetFileName: (filePath: string) => string;
+export declare const GetFileNameWithoutExt: (filePath: string) => string;
+export declare const GetDirectoryPath: (fileName: string) => string;
+export declare const GetWorkingDirectory: () => string;
+export declare const GetDirectoryFiles: (dirPath: string, count: number) => number;
+export declare const ClearDirectoryFiles: () => void;
```

Comment:
While I really like this change, the only issue I find is that we now have to have duplicate definitions of the functions. One to define in the Node package, and then another for TypeScript.

If there is a way to have it so that it auto-discovered them, then that'd be great. Otherwise, it will make maintenance more difficult.

Reply:
To my knowledge I don't believe there is a way to auto discover with regular napi, but I could be wrong. In this latest batch of commits I changed the way these are declared because I was being an idiot about it. Now there is only one declaration of the function (that gotten from the bindings) and the declarations are all applied retroactively.